bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/InsteonBinding.java

```diff
@@ -0,0 +1,518 @@
+ * messages are received) leads to complicated state management because there is no guarantee at what
+ * point (if at all) the binding configuration will be available. It gets even more difficult when
+ * items are created, destroyed, and modified while the binding runs.
+ *
+ * For the above reasons, devices are only instantiated when they are referenced by binding information.
+ * As nice as it would be to discover devices and their properties dynamically, we have abandoned that
+ * path because it had led to a complicated and fragile system which due to the technical limitations
+ * above was inherently squirrely.
+ *
+ *
+ * @author Bernd Pfrommer - Initial contribution
+ * @author Daniel Pfrommer - OpenHAB 1 insteonplm binding
+ * @author Rob Nielsen - Port to OpenHAB 2 insteon binding
+ */
+@NonNullByDefault
+public class InsteonBinding {
+    private final Logger logger = LoggerFactory.getLogger(InsteonBinding.class);
+
+    private Driver m_driver = new Driver();
+    private ConcurrentHashMap<InsteonAddress, InsteonDevice> m_devices = new ConcurrentHashMap<InsteonAddress, InsteonDevice>();
+    private ConcurrentHashMap<String, InsteonChannelConfiguration> m_bindingConfigs
+         = new ConcurrentHashMap<>();
+    private PortListener m_portListener = new PortListener();
+    private int m_devicePollInterval = 300000; // in milliseconds
+    private int m_deadDeviceTimeout = -1;
+    private int m_deviceStatisticsInterval = 600; // in seconds
+    private int m_messagesReceived = 0;
+    private boolean m_isActive = false; // state of binding
+    private int m_x10HouseUnit = -1;
+    private InsteonNetworkHandler m_handler;
+
+    public InsteonBinding(InsteonNetworkHandler handler, String port, @Nullable Integer deviceStatisticsInterval,
+            @Nullable Integer deviceDeadCount, @Nullable Integer devicePollInterval, @Nullable String additionalDevices,
+            @Nullable String additionalFeatures, @Nullable Integer modemDbRetryTimeout) {
+
+        this.m_handler = handler;
+
+        if (deviceStatisticsInterval != null) {
+            m_deviceStatisticsInterval = deviceStatisticsInterval;
+            logger.info("device statistics interval set to {} seconds", m_deviceStatisticsInterval);
+        }
+
+        int deadDeviceCount = 10;
+        if (deviceDeadCount != null) {
+            deadDeviceCount = deviceDeadCount;
+            logger.info("device dead co- Initial contributionunt set to {}", deadDeviceCount);
+        }
+
+        if (devicePollInterval != null) {
+            m_devicePollInterval = devicePollInterval;
+            logger.info("device poll interval set to {} milliseconds", m_devicePollInterval);
+        }
+
+        if (additionalDevices != null) {
+            try {
+                DeviceTypeLoader.s_instance().loadDeviceTypesXML(additionalDevices);
+                logger.info("read additional device definitions from {}", additionalDevices);
+            } catch (Exception e) {
```

Comment:
Does this throw a generic `Exception` or can it be the specific checked exception or `RuntimeException`?

Reply:
I will update to used specific exceptions.